openwisp_monitoring/device/admin.py

```diff
@@ -145,6 +160,20 @@ def get_form(self, request, obj=None, **kwargs):
         )
         return super().get_form(request, obj, **kwargs)
 
+    def get_fields(self, request, obj=None):
+        if not obj or obj.monitoring.status != 'problem':
+            return self.fields
```

Comment:
when working with the openwisp django admin, especially DeviceAdmin which is extended by different modules, never ever brutally override a method like you're doing here: **you're throwing away several customizations that have been done elsewhere**.

You have to always call `super()`, get the result, and do your operations on that result.

Here's what how the add device page looked before your changes:

![Screenshot from 2020-08-20 21-15-07](https://user-images.githubusercontent.com/841044/90844977-c31fe380-e32a-11ea-9896-baba897407ce.png)

After these changes, fields which should not be shown in the add page are instead shown (uuid, management ip, health status):

![Screenshot from 2020-08-20 21-14-33](https://user-images.githubusercontent.com/841044/90845011-d468f000-e32a-11ea-9816-696f4e5dd287.png)

Please pay attention to this! If I didn't notice now we'd introduce another bug.

As you can see, even if the test coverage of each module is pretty high, passing tests + high coverage is not a guarantee of being bug free. Manual testing is **very** important.